test/test_torch.py

```diff
@@ -12320,8 +12320,12 @@ def test_acosh_domain(self, device):
     def test_atanh_domain(self, device):
         # Domain of atanh is (-1, 1)
         sample = torch.tensor([-1.00, 1.00, -1.23, 1.06], device=device)
-        self.assertEqual(torch.isnan(torch.atanh(sample)),  torch.BoolTensor([1, 1, 1, 1]))
-        self.assertEqual(torch.isnan(sample.atanh()), torch.BoolTensor([1, 1, 1, 1]))
+        # For values not in domain (except -1.0 and 1.0), atanh should return nan
+        self.assertEqual(torch.isnan(torch.atanh(sample)),  torch.BoolTensor([0, 0, 1, 1]))
+        self.assertEqual(torch.isnan(sample.atanh()), torch.BoolTensor([0, 0, 1, 1]))
+        # For values -1.0 and 1.0, atanh should return -inf and inf respectively
+        self.assertEqual(torch.isinf(torch.atanh(sample)), torch.BoolTensor([1, 1, 0, 0]))
+        self.assertEqual(torch.isinf(sample.atanh()), torch.BoolTensor([1, 1, 0, 0]))
```

Comment:
For `atanh`, values `-1.0` and `1.0` return `inf`, and for other values not in domain, it returns `nan`. I've also included a check for in-place op.